File: SchoolWebApp-API/Project.Core/Entities/CBE/Cocurriculum/StudentCoCurriculumScore.cs
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.CBE.Cocurriculum
{
    public class StudentCoCurriculumScore: Base
    {
        public int StudentCoCurriculumActivityId { get; set; }
        public StudentCoCurriculumActivity? StudentCoCurriculumActivity { get; set; } = null;

        public int CoCurriculumScoreId { get; set; }
        public CoCurriculumScore? CoCurriculumScore { get; set; } = null;
        public string? Description { get; set; } = null;
    }
}

File: SchoolWebApp-Client/src/app/cbe/cocurriculum/models/student-co-curriculum-score.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class StudentCoCurriculumScore extends ResourceModel<StudentCoCurriculumScore> {
    public studentCoCurriculumActivityId?: number;
    public studentCoCurriculumActivity?: any;
    public coCurriculumScoreId?: number;
    public coCurriculumScore?: any;
    public description?: string;

    constructor(model?: Partial<StudentCoCurriculumScore>) {
        super(model);
    }
}
